packages/@lwc/compiler/src/options.ts

```diff
@@ -84,6 +84,7 @@ export interface TransformOptions {
     customRendererConfig?: CustomRendererConfig;
     enableLwcSpread?: boolean;
     disableSyntheticShadowSupport?: boolean;
+    instrumentation?: CompilerInstrumentation;
```

Comment:
I'm passing the instrumentation object through the configs because:
- I didn't like introducing `metrics` or `instrumentation` as a formal API on the TransformResult interface or `compile()` itself. Metrics and instrumentation feel like they should be part of some global execution state of the compiler (similar to `console.xxxx`).
- it keeps the pattern consistent between the template compiler, JS compiler, and style compiler in treating the instrumentation object as something configurable by the compiler's user.

Reply:
This also makes it so we don't collect any instrumentation unless the person using the compiler wants to and has set an instrumentation object on the config.

An alternative would be to make this a `boolean` and then pass the resulting instrumentation object as part of the Result interface; that was rejected because I didn't want to make it part of the result interface, and also wanted to give the ability to provide a custom instrumentation object from outside the compiler.